src/compiler/scanner.ts

```diff
@@ -858,28 +858,76 @@ namespace ts {
             }
         }
 
+        function scanNumberFragment(): string {
+            let start = pos;
+            let allowSeperator = false;
+            let result: string;
+            while (true) {
+                const ch = text.charCodeAt(pos);
+                if (allowSeperator && ch === CharacterCodes._) {
+                    allowSeperator = false;
+                    tokenFlags |= TokenFlags.ContainsSeparator;
+                    result = (result || "") + text.substring(start, pos);
+                    pos++;
+                    start = pos;
+                    continue;
+                }
+                if (isDigit(ch)) {
+                    allowSeperator = true;
+                    pos++;
+                    continue;
+                }
+                break;
+            }
+            if (text.charCodeAt(pos - 1) === CharacterCodes._) {
+                pos--;

```

Comment:
We do this exact set of three lines multiple times in this PR. Would it make sense to provide an optional argument to `error` to override the position it uses?